Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/ByMagDurCurve.tsx

```diff
@@ -22,110 +22,130 @@
 //******************************************************************************************************
 
 import * as React from 'react';
-import { ReactTable } from '@gpa-gemstone/react-table';
-import * as _ from 'lodash';
+import CurveForm from './CurveForm';
+import GenericByPage from '../CommonComponents/GenericByPage';
 import { Application, OpenXDA } from '@gpa-gemstone/application-typings';
-import { SearchBar, Search, Modal, Warning } from '@gpa-gemstone/react-interactive';
-import { useAppSelector, useAppDispatch } from '../hooks';
-import { MagDurCurveSlice } from '../Store/Store';
+import { Modal, Warning } from '@gpa-gemstone/react-interactive';
+import { SystemCenter } from '../global';
 import { CrossMark } from '@gpa-gemstone/gpa-symbols';
-import CurveForm from './CurveForm';
-
 
 declare var homePath: string;
+const controllerPath = `${homePath}api/SystemCenter/StandardMagDurCurve`
 const emptyCurve: OpenXDA.Types.MagDurCurve = {
     ID: 0,
     Name: 'Curve',
     Area: '0.00001 0, 1000 0, 1000 1, 0.00001 1, 0.00001 0',
     Color: ''
 };
+const fieldCols: SystemCenter.IByCol<OpenXDA.Types.MagDurCurve>[] = [
+    { Field: 'Name', Label: 'Name', Type: 'string', Width: 'auto' }
+]
 
-const ByMagDurCurve: Application.Types.iByComponent = (props) => {
-    let dispatch = useAppDispatch();
-
-    const cState = useAppSelector(MagDurCurveSlice.SearchStatus);
-    const data = useAppSelector(MagDurCurveSlice.SearchResults);
-
-    const sortKey = useAppSelector(MagDurCurveSlice.SortField);
-    const filters = useAppSelector(MagDurCurveSlice.SearchFilters);
-    const ascending = useAppSelector(MagDurCurveSlice.Ascending);
-
+const ByMagDurCurve: Application.Types.iByComponent = () => {
     const [curve, setCurve] = React.useState<OpenXDA.Types.MagDurCurve>(emptyCurve);
-
+    const [errors, setErrors] = React.useState<string[]>([]);
+    const [newEdit, setNewEdit] = React.useState<Application.Types.NewEdit>('Edit');
     const [showModal, setShowModal] = React.useState<boolean>(false);
     const [showDelete, setShowDelete] = React.useState<boolean>(false);
-    const [errors, setErrors] = React.useState<string[]>([]);
+    const [refreshCount, refreshData] = React.useState<number>(0);
+
+    function addMagDurCurve() {
+        let handle = $.ajax({
+            type: "POST",
+            url: `${homePath}api/SystemCenter/StandardMagDurCurve/Add`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify(curve),
+            dataType: "json",
+            cache: false,
+            async: true
+        }).done(() => {
+            refreshData(x => x + 1);
+        })
```

Comment:
Unless there is something non-standard about this call, can we use the GenericController class to achieve this? That way, if we ever have to change this function we don't have to change it in a bunch of individual places. You do this in more places that just here, can all of them be changed?